cli/azd/resources/starter/azure.yaml

```diff
@@ -0,0 +1,29 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/v1.0/azure.yaml.json
+
+# This is an example starting azure.yaml file containing several services.
+# Make changes as needed to describe your application setup.
+
+# Name of the application.
+name: azd-starter
+services:
+  # An example for a python API service.
+  # The service is named 'python-api'.
+  # The language is 'python'.
+  # The source code is located in the project (azure.yaml) directory.
+  # The service will be hosted on Azure App Service.
+  python-api:
```

Comment:
Could we scaffold an infra/ with an empty main.bicep and a main.tf, and a README in the infra/. 

The README could be very basic and just 1) explain that you can delete either the main.bicep or the main.tf depending on you IaC provider preference and 2) link the user to an example todo template infra/ and maybe [our devhub docs](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/make-azd-compatible?pivots=azd-create) so they get a feel for what's needed.

I only mention this because the only real requirements of an azd template today are the azure.yaml and an infra/ with a main.tf or main.bicep. It'd be nice if we could provide a bit more guidance here.

Reply:
`main.bicep` has been added with the latest push. I'll work on `main.tf` next.

Reply:
Sweet - thank you!

Reply:
Can we make sure when `azd init` is run for a project that already have e.g., /infra (and main.bicep etc. in the folder), `azd init` will not clobber what's already there? CSU has been modeling all their app templates based on azd conventions. So they have /.github/workflow, /infra, /src etc. folders.

Reply:
@savannahostrowski I think the feedback from PC here is valid. I think we maybe want a separate template that has these extra files included, and a minimal one with only `azure.yaml` (the one we currently call `Empty`). 

It sounds like we also want Terraform separately, so the list looks like:
- `Empty Template` (current) Or maybe `Minimal starter`? Or `Minimal template`?
- `Basic starter (bicep)` (add)
- `Basic starter (terraform)` (add)

Or maybe like you said, adding an additional prompt:

- `Basic starter` - with the prompt: `Would you like to use bicep or terraform as your provider?`

Personally, I'd lean towards less steps and having bicep and terraform separate. I also wonder if we need a message somewhere to say "terraform is in alpha ....". If you can help provide feedback on UX here, I can get the issue updated.

Reply:
Thinking more about this, let's go with option 1 (show multiple basic starters). Fewer steps + providing options = win IMO.

@gkulin We should add some details in the docs to outline these new options.

https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/make-azd-compatible?pivots=azd-create

Reply:
From savannah, we agreed upon:

- `Starter - Bicep (azd-bicep-starter)` 
- `Starter - Terraform (azd-bicep-terraform)`